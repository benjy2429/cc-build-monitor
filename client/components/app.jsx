import React from 'react';
import ProjectWrapper from './project-wrapper';
import loader from './loader';

class App extends React.PureComponent {
  render() {
    const { data: projects } = this.props;
    return (
      <div className="full-height">
        <ProjectWrapper projects={projects} />
      </div>
    );
  }
}

export default loader(App);
export { App as UnwrappedApp };
