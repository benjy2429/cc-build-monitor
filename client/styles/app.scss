@import 'project';

html,
body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
  background-color: #161616;
  color: #fff;
}

html,
body,
#app,
.full-height {
  height: 100%;
  width: 100%;
}

#app {
  margin: 20px;
}

.project-wrapper {
  display: flex;
  flex-flow: column wrap;
  justify-content: center;
  min-height: 100%;
}

.status-row {
  flex: 1;
  margin: 20px 0;

  .status-heading {
    font-size: 50px;
    margin: 0 0 10px;
  }

  .status-projects {
    display: flex;
    flex-flow: row wrap;
  }
}
