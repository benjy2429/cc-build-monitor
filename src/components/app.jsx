import React from 'react';

export default class App extends React.Component {
    render() {
        return (
            <html>
                <head>
                    <title>CircleCI Build Monitor</title>
                    <link rel="stylesheet" href="/assets/stylesheet.css" />
                </head>
                <body>
                    <h1>Hello World!</h1>
                </body>
            </html>
        );
    }
};
