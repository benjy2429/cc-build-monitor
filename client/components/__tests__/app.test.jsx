import React from 'react';
import { shallow } from 'enzyme';
import { UnwrappedApp as App } from '../app';

const defaultProps = {
  data: [
    { name: 'project1' },
    { name: 'project2' },
  ],
};

describe('Loader', () => {
  it('renders correctly', () => {
    const component = shallow(
      <App {...defaultProps} />,
    );
    expect(component).toMatchSnapshot();
  });
});
