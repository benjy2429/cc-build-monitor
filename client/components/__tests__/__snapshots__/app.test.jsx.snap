// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders correctly 1`] = `
<div
  className="full-height"
>
  <ProjectWrapper
    projects={
      Array [
        Object {
          "name": "project1",
        },
        Object {
          "name": "project2",
        },
      ]
    }
  />
</div>
`;
